File: samples/java/SupportChat/Server/Support/src/main/java/systems/zlink/samples/supportchat/server/support/spots/conversationspot/SendChatMessageHandler.java
package systems.zlink.samples.supportchat.server.support.spots.conversationspot;

import systems.zlink.framework.ZLinkMessageContext;
import systems.zlink.framework.spots.ZLinkSpotActorRequestHandler;
import systems.zlink.samples.supportchat.server.support.actors.SupportUserActor;
import systems.zlink.samples.supportchat.shared.contracts.Messages;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public final class SendChatMessageHandler
        implements ZLinkSpotActorRequestHandler<
                ConversationSpot,
                SupportUserActor,
                Messages.SendChatMessageReq,
                Messages.SendChatMessageRes> {
    @Override
    public CompletionStage<Messages.SendChatMessageRes> handle(
            ConversationSpot spot,
            SupportUserActor actor,
            ZLinkMessageContext context,
            Messages.SendChatMessageReq request) {
        return CompletableFuture.completedFuture(spot.sendMessage(actor, request));
    }
}

File: samples/kotlin/SupportChat/Server/Support/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/support/infrastructure/zlink/spots/conversationspot/handlers/SendChatMessageHandler.kt
package systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.handlers

import systems.zlink.framework.ZLinkMessageContext
import systems.zlink.framework.kotlin.ZLinkSuspendingSpotActorRequestHandler
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.actors.SupportUserActor
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.ConversationSpot
import systems.zlink.samples.kotlin.supportchat.shared.contracts.SendChatMessageReq
import systems.zlink.samples.kotlin.supportchat.shared.contracts.SendChatMessageRes

class SendChatMessageHandler :
    ZLinkSuspendingSpotActorRequestHandler<
        ConversationSpot,
        SupportUserActor,
        SendChatMessageReq,
        SendChatMessageRes,
    > {
    override suspend fun handle(
        spot: ConversationSpot,
        actor: SupportUserActor,
        context: ZLinkMessageContext,
        request: SendChatMessageReq,
    ): SendChatMessageRes = spot.sendMessage(actor, request)
}
